File: clojure/nth-prime/_iterations/1/src/nth_prime.clj
(ns nth-prime)

(defn prime?
  "Determines if n is prime."
  [n]
  (or (= n 2) (every? #(not= (mod n %) 0) (range 2 (inc (Math/sqrt n))))))

(defn nth-prime
  "Returns the prime number in the nth position."
  [n]
  (when (not (pos? n))
    (throw (IllegalArgumentException.)))
  (->> (iterate inc 2)
       (filter prime?)
       (take n)
       last))

File: ruby/nth-prime/_iterations/1/nth_prime.rb
module Prime
  def self.nth(n)
    raise ArgumentError, "n must be positive" unless n.positive?

    primes = []
    x = 2
    while primes.size < n
      primes << x if primes.none? { |p| x % p == 0 }
      x += 1
    end
    primes.last
  end
end
